pkg/config/analysis/analyzers/virtualservice/destinationrules.go

```diff
@@ -116,10 +116,18 @@ func initDestHostsAndSubsets(ctx analysis.Context) map[hostAndSubset]bool {
 	ctx.ForEach(collections.IstioNetworkingV1Alpha3Destinationrules.Name(), func(r *resource.Instance) bool {
 		dr := r.Message.(*v1alpha3.DestinationRule)
 		drNamespace := r.Metadata.FullName.Namespace
-
+		drHost := dr.GetHost()
+		if drNamespace == util.IstioSystemNamespace {
+			// Convert the short name host to FQDN, such as convert "review" to "review.default.svc.cluster.local"
+			// using "default" as the namespace of host if destination rule's namespace is "istio-system"
```

Comment:
I am not sure if this is entirely correct.  We can't assume default as namespace imho.

Per the spec `Note for Kubernetes users: When short names are used (e.g. “reviews” instead of “reviews.default.svc.cluster.local”), Istio will interpret the short name based on the namespace of the rule, not the service. A rule in the “default” namespace containing a host “reviews” will be interpreted as “reviews.default.svc.cluster.local”, irrespective of the actual namespace associated with the reviews service. To avoid potential misconfigurations, it is recommended to always use fully qualified domain names over short names.`, I think we should not allow short name as host when DR is deployed to istio-system.

Reply:
Hi @linsun, it's okay that only FQDN as the host when DR is deployed to istio-system, so there should be a warning in this anaysis if short name are used. Besides that, I think the docs also need to modify as well.

Reply:
new submit, done!

Reply:
What does Pilot do? We can't not allow short name host in the analyzer if pilot allows it or we will get incorrect messaging.

This is the fundamental flaw with all analyzers IMO, we imperfectly recreate pilot's logic. Would be better to use it directly

Reply:
Besides the existing resource, such as virtual service and destination rule, user may use analyzer to do an analysis based on their yaml files, etc. 
So what's the relationship between pilot and analyzer?  Is the latter just a double insurance for the former or both of them are co-work together?